### DISCUSSION TITLE
ERROR running "server_async.py" in "async_io.py"

### DISCUSSION THREAD

**Discussion by SeanMcKinnon** [pymodbus-dev/pymodbus#1896] [Help] [answered]
Hello, when I try to run the example server (server_async.py) I'm getting an error that **self.server.framer** in "async_io.py" is a string and isn't callable as a function. The server starts, and "listens" but crashes right after the client connects.
Any ideas what would be the cause? I downgraded my python version from the latest as well as tested an earlier release of pymodbus (started with 3.5.4), but got the same error. For now I'm using the REPL, but would like to use the "server_async.py". Thanks!

By the way, I've used this package many times and it's great! Thanks for all the hard work. 

TypeError: 'str' object is not callable

Line 63 in "async_io.py"
  self.running = True
            self.framer = **self.server.framer**(
                self.server.decoder,
                client=None,
            )

**Comment by janiversen** [comment]
This could be a bug, let me test later.

**Comment by janiversen** [comment]
When you downgraded to v3.5.4 did you remember to also downgrade the example ?

The problem might be that we do not upload altered examples to dev on read the docs.

**Comment by janiversen** [comment]
You do not use the examples that maches the release:

<img width="2053" alt="Captura de pantalla 2023-11-30 a las 14 28 16" src="https://github.com/pymodbus-dev/pymodbus/assets/6185391/ff482100-f413-4630-a200-73e7ead4ffe2">

As you can see server_async works well on dev.

But as I wrote earlier the problem can be, that we do not upload examples for dev with every commit.

**Comment by SeanMcKinnon** [comment, marked as answer]
I tried 3.5.4 again making sure to use the correct example and everything works! I must have mixed something up. Thanks!

**Comment by janiversen** [comment]
No problem.
